### DISCUSSION TITLE
Can you test tweepy in a Sandboxed twitter environment

### DISCUSSION THREAD

**Discussion by ghost** [tweepy/tweepy#1903] [Questions] [answered]
So I'm working on a  DEV / ACC / PROD based twitter bot project.

I want to test is everything works in DEV and ACC before deploying to PROD.

But how can I create_tweet in a "sand-boxed" or "test" twitter environment.

I do't really want to tweet in DEV and ACC but surely want to test if things are working 

Thank you

**Comment by Harmon758** [comment, marked as answer]
With Elevated access, you can have 3 App environments (development, staging, production) per Project.

https://developer.twitter.com/en/docs/twitter-api/getting-started/about-twitter-api
https://blog.twitter.com/developer/en_us/topics/tools/2021/build-whats-next-with-the-new-twitter-developer-platform

You'll still need an account to Tweet with. Otherwise, I'm not sure how you expect to test Tweeting without actually Tweeting or what "environment" you're asking for. At the very least, you would need to record (or mock) a response from an actual call to the API to replay, like Tweepy does for its tests.
